packages/terra-dialog/src/Dialog.jsx

```diff
@@ -38,11 +40,13 @@ const Dialog = ({ header, footer, onClose, children, ...customProps }) => {
     'dialog',
   ]);
 
-  const actionHeader = <div className={cx('dialog-header')}><ActionHeader title={header} onClose={onClose} /></div>;
+  const closeButton = onClose ? <div className={cx('dialog-header-close')}><Button icon={<IconClose />} aria-label="Close Button" onClick={onClose} isIconOnly /></div> : null;
```

Comment:
UX has requested that the icon be themable. Do to this, we are making the icon a background image on span that is nested inside the button.

Take a look at how @msaikaushik1992 is implementing the close icon in popup.
* [Setting the span inside the button which we'll set a background image on](https://github.com/cerner/terra-core/blob/7709d89ced0531985881f2675b9ec896eaf34152/packages/terra-popup/src/_PopupContent.jsx#L122-L123)
* [Adding styles to set background image by inlining SVG with dataURI](https://github.com/cerner/terra-core/blob/7709d89ced0531985881f2675b9ec896eaf34152/packages/terra-popup/src/PopupContent.scss#L31-L42)

Reply:
@saedar May need to talk to @neilpfeiffer. Sounds like this may change some.

Reply:
Preliminary: https://github.com/cerner/terra-core/pull/1250/commits/26e7f056a00ff51fe00bffc670a7d53f9db26c6d

I'll follow up with Neil.